src/query.ts

```diff
@@ -620,4 +627,20 @@ export interface RunQueryInfo {
     | 'MORE_RESULTS_AFTER_LIMIT'
     | 'MORE_RESULTS_AFTER_CURSOR'
     | 'NO_MORE_RESULTS';
+  explainMetrics?: ExplainMetrics;
+}
+
+export interface ExplainMetrics {
+  planSummary?: PlanSummary;
+  executionStats?: ExecutionStats;
+}
+export interface ExecutionStats {
+  resultsReturned?: number;
+  executionDuration?: google.protobuf.IDuration;
+  readOperations?: number;
+  debugStats?: JSONValue;
```

Comment:
The client library doesn't typically return `Map` objects. It usually returns dictionaries when an object needs to be returned that maps keys to values. For example, an interface like `{ [key: string]: any };`.

The issue here is that the values coming from the server are of type `google.protobuf.IStruct` and the `serializer.toProto3JSON` function is used to decode these types returning a `JSONValue` type which is stored in debugStats. The compiler only guarantees the value will be a `JSONValue`, but in practice a `{ [key: string]: any };` type should always be returned.

We can change the return type to `{ [key: string]: any };` and the compiler won't complain because we are using `Object.assign` so maybe we should do that. What do you think?

Reply:
I think that works, would like a NodeJS opinion however :)